src/libs/dynamodb-lib.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  DynamoDBClient,
+  PutItemCommand,
+  GetItemCommand,
+  ScanCommand,
+} from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
+import { sendMetricData } from "./cloudwatch-lib";
+import { marshall, unmarshall } from "@aws-sdk/util-dynamodb";
+
+const client = new DynamoDBClient({ region: process.env.region });
+
+export async function putItem({
+  tableName,
+  item,
+}: {
+  tableName: string;
+  item: { [key: string]: any };
+}) {
+  const params = {
+    TableName: tableName,
+    Item: marshall(item, {
+      removeUndefinedValues: true,
+    }),
+  };
+
+  try {
+    if (item && item.id) console.log(`Putting item with id: ${item.id}:`);
+
+    const command = new PutItemCommand(params);
+    const result = await client.send(command);
+    if (item && item.id)
+      console.log(
+        `Record processed for item: ${item.id}:`,
+        JSON.stringify(result, null, 2)
+      );
+
+    await sendMetricData({
+      Namespace: process.env.namespace,
+      MetricData: [
+        {
+          MetricName: `${tableName}_dynamo_updates`,
+          Value: 0,
+        },
+      ],
+    });
+
+    return result;
+  } catch (error) {
+    console.error("ERROR updating record in dynamodb: ", error);
+    await sendMetricData({
+      Namespace: process.env.namespace,
+      MetricData: [
+        {
+          MetricName: `${tableName}_dynamo_updates`,
+          Value: 1,
+        },
+      ],
+    });
+    return;
+  }
+}
+
+export async function getItem({
+  tableName,
+  id,
+}: {
+  tableName: string | undefined;
+  id: string;
+}) {
+  const item = (
+    await client.send(
+      new GetItemCommand({
+        TableName: tableName,
+        Key: {
+          id: {
+            S: id,
+          },
+        },
+      })
+    )
+  ).Item;
+  if (!item) return null;
+
+  /* Converting the DynamoDB record to a JavaScript object. */
+  return unmarshall(item);
```

Comment:
```suggestion
export async function getItem({
  tableName,
  key,
}: {
  tableName: string;
  key: {
      [key: string]: NativeAttributeValue;
    };
}) {

const getItemCommandInput: GetItemCommandInput = {
    TableName: tableName,
    Key: marshall(key),
  };
  
  const item = (await client.send(new GetItemCommand(getItemCommandInput)))
    .Item;
  if (!item) return null;

  /* Converting the DynamoDB record to a JavaScript object. */
  return unmarshall(item);
```

this is a recent addition to compare, which is where im sure you got this logic. Lets use the entire key here instead of just the id. this will allow us to use PK and SK for our table and cleans things up quite a bit with the types